### PR TITLE
Implement couch_log_devnull for suppressing logging during testing

### PR DESCRIPTION
**Pull request by jaydoane**
COUCHDB-2899

### CODE REVIEW

File: src/couch_log_devnull.erl [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    write_log("[error] " ++ Fmt, Args).
+
+critical(Fmt, Args) ->
+    write_log("[critical] " ++ Fmt, Args).
+
+alert(Fmt, Args) ->
+    write_log("[alert] " ++ Fmt, Args).
+
+emergency(Fmt, Args) ->
+    write_log("[emergency] " ++ Fmt, Args).
```
Comment (kxepal) [inline src/couch_log_devnull.erl]: Why construct another list if the result doesn't matters?

**Comment by kxepal** [comment]
Indeed, that's the easiest way to handle this, but I think it's not right one. In CouchDB 1.x we had special log level `none` which disables all the logging. And this one is correct: you shouldn't jungle backends to toggle logging activity - it's all the matter of logging level, where there is the highest one that suppress everything. 

However, this will require more work to do and allow to configure couch_log via config. That's not bad, since that's need to do in anyway. Would you like to handle this?

File: src/couch_log_devnull.erl (hunk 1 above)
Comment (jaydoane) [inline src/couch_log_devnull.erl]: Good point. Obviously, I just copied this from couch_log_stderr to minimize code diffs, but I'll get rid of pointless operations if you intend to eventually merge it.

**Comment by jaydoane** [comment]
I can certainly look into configuring couch_log via config, but right now my priority is re-enabling the test suite for downstream CI builds. This is just a quick-and-dirty hack that would make my life a little easier for the time being, but I can understand not wanting it in couchdb.

**Comment by kxepal** [comment]
@jaydoane ok, so I just submitted https://github.com/apache/couchdb-couch-log/pull/10
To solve your problem there is nothing need to do since [eunit.ini](https://github.com/apache/couchdb/blob/master/rel/files/eunit.ini) already contains magic option that turns off any logging so these calls will not reach the backend whatever it used.

**Comment by jaydoane** [comment]
@kxepal, excellent. Thank you! Looks like you'll need to add .travis.yml in your commit, though.

**Comment by kxepal** [comment]
@jaydoane yea, I should add travis.yml everywhere. Thanks for reminder (:

**Pull request closed without merging**